app/commands/todo/abandon.rb

```diff
@@ -7,12 +7,17 @@ module Abandon
         class Command
           attr_reader :payload, :aggregate_id
 
+          def self.build(**args)
+            new(**args).tap(&:validate)
+          end
+
           def initialize(params)
             @payload = params.slice(:todo_id, :abandoned_on)
             @aggregate_id = payload.delete(:todo_id)
           end
 
-          def valid?
+          def validate
+            raise BadRequest, 'todo_id is blank' if aggregate_id.nil?
```

Comment:
Do we need an error class with a more specific name? Something like `InvalidCommandPayload` or `InvalidCommand`?